src/django_components/templatetags/component_tags.py

```diff
@@ -57,34 +136,191 @@ def _component_dependencies(type: Literal["js", "css"]) -> SafeString:
             f"Unknown dependency type in {{% component_dependencies %}}. Must be one of 'css' or 'js', got {type}"
         )
 
-    return mark_safe(placeholder)
+    return TextNode(mark_safe(placeholder))
 
 
-@register.simple_tag(name="component_css_dependencies")
```

Comment:
I've changed `component_dependencies` tags to use `@register.tag()` instead of `@register.simple_tag()`, so that the documentation could be auto-generated.